internal/ignition/ignition.go

```diff
@@ -388,6 +390,12 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte,
 		return err
 	}
 
+	//lock the critical section where the files are writen to the installer
+	//path to avoid deletion of its contect mid air
+	r.Lock()
```

Comment:
is it possible to "lose" the object before this line (``r.Lock()``) gets the chance to get executed?

Reply:
Actually yes and it is a good catch. I changed the code so the lock is released only at the end of the flow.